java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/JavaCheckVerifier.java

```diff
@@ -107,32 +111,33 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
 
   private void collectExpectedIssues(String comment, int line) {
     if (comment.startsWith(TRIGGER)) {
-      comment = StringUtils.remove(comment, TRIGGER);
+      String cleanedComment = StringUtils.remove(comment, TRIGGER);
 
-      String expectedMessage = StringUtils.substringBetween(comment, "{{", "}}");
+      String expectedMessage = StringUtils.substringBetween(cleanedComment, "{{", "}}");
+      int expectedLine = line;
 
-      comment = StringUtils.stripEnd(StringUtils.remove(comment, "{{" + expectedMessage + "}}"), " \t");
-      if (StringUtils.startsWith(comment, "@")) {
+      cleanedComment = StringUtils.stripEnd(StringUtils.remove(cleanedComment, "{{" + expectedMessage + "}}"), " \t");
+      if (StringUtils.startsWith(cleanedComment, "@")) {
         final int lineAdjustment;
-        final char firstChar = comment.charAt(1);
-        final int endIndex = comment.indexOf(' ');
+        final char firstChar = cleanedComment.charAt(1);
+        final int endIndex = cleanedComment.indexOf(' ');
         if (endIndex == -1) {
-          lineAdjustment = Integer.parseInt(comment.substring(2));
-          comment = "";
+          lineAdjustment = Integer.parseInt(cleanedComment.substring(2));
+          cleanedComment = "";
         } else {
-          lineAdjustment = Integer.parseInt(comment.substring(2, endIndex));
-          comment = comment.substring(endIndex + 1).trim();
+          lineAdjustment = Integer.parseInt(cleanedComment.substring(2, endIndex));
+          cleanedComment = cleanedComment.substring(endIndex + 1).trim();
         }
         if (firstChar == '+') {
-          line += lineAdjustment;
+          expectedLine += lineAdjustment;
         } else if (firstChar == '-') {
-          line -= lineAdjustment;
+          expectedLine -= lineAdjustment;
         }
```

Comment:
@mpaladin `#` has never been authorized, I simply putted it to show that it's ignored. However, I agree with you and I'll  remove it. We will then only consider `-` and `+`. If any other character is used after the `@`, it will now fail.